### PR TITLE
Haiku support

### PR DESCRIPTION
**Pull request by MaddTheSane**
Based on patches linked in #66 by @pulkomandy.

### CODE REVIEW

File: BasiliskII/src/BeOS/extfs_beos.cpp [hunk 1]
```diff
@@ -458,3 +459,31 @@ bool extfs_rename(const char *old_path, const char *new_path)
...
+	return buffer;
+}
+
+// Convert from MacRoman to host OS filename encoding
+const char *macroman_to_host_encoding(const char *filename)
+{
+	int32 state = 0;
+	static char buffer[512];
+	int32 insize = strlen(filename);
+	int32 size = 512;
```
Comment (asvitkine) [inline BasiliskII/src/BeOS/extfs_beos.cpp]: I think this needs to be 511, else if 512 bytes will be written, line 487 will write pas the buffer.

Please change to:

int32 size = sizeof(buffer) -1;

And also do the same above.

File: BasiliskII/src/BeOS/video_beos.cpp [hunk 2]
```diff
@@ -42,12 +42,16 @@
 #include "readcpu.h"
 #include "newcpu.h"
 
-#define DEBUG 0
+#define DEBUG 1
```
Comment (asvitkine) [inline BasiliskII/src/BeOS/video_beos.cpp]: Keep this at 0?

File: BasiliskII/src/BeOS/video_beos.cpp [hunk 3]
```diff
@@ -98,7 +122,7 @@ class BitmapView : public BView {
 
 class MacWindow : public BDirectWindow {
 public:
-	MacWindow(BRect frame);
+	MacWindow(BRect frame, BeOS_monitor_desc& monitor);
```
Comment (asvitkine) [inline BasiliskII/src/BeOS/video_beos.cpp]: Pass the new param by const ref (i.e. const BeOS_monitor_desc&) please.

File: BasiliskII/src/BeOS/video_beos.cpp [hunk 4]
```diff
@@ -136,7 +161,7 @@ class MacWindow : public BDirectWindow {
 
 class MacScreen : public BWindowScreen {
 public:
-	MacScreen(const char *name, int mode_bit, status_t *error);
+	MacScreen(BeOS_monitor_desc& monitor, const char *name, int mode_bit, status_t *error);
```
Comment (asvitkine) [inline BasiliskII/src/BeOS/video_beos.cpp]: Pass the new param by const ref (i.e. const BeOS_monitor_desc&) please.

File: BasiliskII/src/BeOS/video_beos.cpp [hunk 5]
```diff
@@ -313,8 +415,11 @@ void video_set_palette(uint8 *pal, int num)
  *  Switch video mode
  */
 
-void video_switch_to_mode(const video_mode &mode)
+void BeOS_monitor_desc::switch_to_current_mode()
 {
+	const video_mode &mode = get_current_mode();
+	D(bug("Switching modes!\n"));
+	fprintf(stderr, "width: %d", 1 << mode.depth);
```
Comment (asvitkine) [inline BasiliskII/src/BeOS/video_beos.cpp]: Remove debug message? Or if you want to keep it, it probably needs a \n.

**Comment by asvitkine** [comment]
Thanks! Left a few comments, otherwise looks good.

**Comment by pulkomandy** [comment]
thanks for the review, fixed the issues and made a new PR from my own fork: https://github.com/cebix/macemu/pull/68

**Comment by MaddTheSane** [comment]
As @pulkomandy has his own pull request, I'm closing this one.

**Pull request closed without merging**